mods/edge-hot-corner-desktop-switch.wh.cpp

```diff
@@ -0,0 +1,173 @@
+// ==WindhawkMod==
+// @id              edge-hot-corner-desktop-switch
+// @name            Edge Hot-Corners Desktop Switch
+// @description     Switches virtual desktops when the mouse hovers at the left or right screen edge for a moment.
+// @version         0.1
+// @author          Sanskar Prasad
+// @github          https://github.com/sanskarprasad
+// @homepage        https://your-personal-homepage.example.com/edge-hot-corner-desktop-switch
+// @include         explorer.exe
+// ==/WindhawkMod==
+
+// ==WindhawkModSettings==
+/*
+- thresholdMs: 1000
+  $name: Time threshold at edge (ms)
+  $description: >
+    Milliseconds the cursor must stay in the hot zone before
+    switching desktops.
+
+- edgeWidth: 5
+  $name: Detection edge width (px)
+  $description: >
+    Width in pixels of the left/right screen hot zone.
+*/
+// ==/WindhawkModSettings==
+
+#include <Windows.h>
+
+// #include <windhawk-sdk.h>
+
+struct {
+    int thresholdMs;
+    int edgeWidth;
+} settings;
+
+volatile bool g_Running = false;
+HANDLE g_hThread = nullptr;
+
+void LoadSettings() {
+    settings.thresholdMs = Wh_GetIntSetting(L"thresholdMs");
+    settings.edgeWidth = Wh_GetIntSetting(L"edgeWidth");
+}
+
+void SimulateDesktopSwitch(int direction) {
+    // direction: -1 = previous (Win+Ctrl+Left), +1 = next (Win+Ctrl+Right)
+    WORD vkDir = (direction < 0 ? VK_LEFT : VK_RIGHT);
+    INPUT inputs[6] = {};
+
+    // Win down (extended)
+    inputs[0].type = INPUT_KEYBOARD;
+    inputs[0].ki.wVk = VK_LWIN;
+    inputs[0].ki.dwFlags = KEYEVENTF_EXTENDEDKEY;
+
+    // Ctrl down (not extended)
+    inputs[1].type = INPUT_KEYBOARD;
+    inputs[1].ki.wVk = VK_CONTROL;
+
+    // Arrow down (extended)
+    inputs[2].type = INPUT_KEYBOARD;
+    inputs[2].ki.wVk = vkDir;
+    inputs[2].ki.dwFlags = KEYEVENTF_EXTENDEDKEY;
+
+    // Arrow up  (extended + key-up)
+    inputs[3].type = INPUT_KEYBOARD;
+    inputs[3].ki.wVk = vkDir;
+    inputs[3].ki.dwFlags = KEYEVENTF_EXTENDEDKEY | KEYEVENTF_KEYUP;
+
+    // Ctrl up
+    inputs[4].type = INPUT_KEYBOARD;
+    inputs[4].ki.wVk = VK_CONTROL;
+    inputs[4].ki.dwFlags = KEYEVENTF_KEYUP;
+
+    // Win up   (extended + key-up)
+    inputs[5].type = INPUT_KEYBOARD;
+    inputs[5].ki.wVk = VK_LWIN;
+    inputs[5].ki.dwFlags = KEYEVENTF_EXTENDEDKEY | KEYEVENTF_KEYUP;
+
+    // Send them all in one go
+    UINT sent = SendInput(_countof(inputs), inputs, sizeof(INPUT));
```

Comment:
This won't work if an elevated program is focused, such as Windhawk or Task Manager.
As a workaround, you can set the taskbar as the foreground window before calling `SendInput`:
https://github.com/ramensoftware/windhawk-mods/blob/8c63a08f1a5c540f9061b4a896dfdf18a0bcd843/mods/taskbar-scroll-actions.wh.cpp#L674-L678